### DISCUSSION TITLE
WebHooks from someone else's repository? Is that possible?

### DISCUSSION THREAD

**Discussion by loadaverage** [community/community#22956] [New to GitHub]
<p>I need to receive ‘release’ events from third-party (that not belong to me) repository, so that I can rebuild my Docker images.</p>
<p>After reading <a href="https://developer.github.com/webhooks/" rel="nofollow noopener">https://developer.github.com/webhooks/</a>:</p>
<p>&gt;&gt; Each event corresponds to a certain set of actions that can happen to your organization and/or repository.</p>
<p>So, I can’t receive WebHooks from not mine repositories?</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>No, it isn’t possible to receiive webhooks from an organization or repository that you do not own without an owner’s permission. If you can convince them to install the webhooks for you though, that might be an option.</p>
<p>You can, however, periodically check the releases API for new updates and generate your own events.</p>

**Comment by loadaverage** [comment]
<p>Thanks!</p>
<p>I think, pub/sub functionality on GitHub will be very cool thing.</p>

**Comment by standbyme** [comment]
<p>Oh, I have met the same question these days.</p>
<p>Do you choose <em>Polling now?</em></p>

**Comment by GottZ** [comment]
<p>i agree. a subscription hook would be awesome to have. i currently have the same problem in front of me. i just want to set it as trigger for building docker images and auto deployment.</p>

**Comment by puteulanus** [comment]
<p>I use <a href="http://feed43.com/" rel="nofollow noopener">feed43</a> to build RSS for the repo, then use <a href="https://ifttt.com/" rel="nofollow noopener">IFTTT</a> to target simple webhook by RSS.</p>
<p>To call some complicate API like <a href="http://quay.io" rel="nofollow noopener">quay.io</a>’s build target, I write a <a href="https://www.google.com/script/start/" rel="nofollow noopener">Google Script</a> and call it by IFTTT’s webhook.</p>
<p>It works well for my personal project. Hope it can help you.</p>

**Comment by BoxedFruits** [comment]
Has anyone tried using a WebSub? Not sure if it was supported when this post was originally made.

**Comment by gauravGo7** [comment]
No, it isn’t possible to receiive webhooks from an organization or repository that you do not own without an owner’s permission. If you can convince them to install the webhooks for you though, that might be an option.

You can, however, periodically check the releases API for new updates and generate your own events...

**Comment by gauravGo7** [comment]
No, it isn’t possible to receiive webhooks from an organization or repository that you do not own without an owner’s permission. If you can convince them to install the webhooks for you though, that might be an option.

You can, however, periodically check the releases API for new updates and generate your own events.

**Reply by BoxedFruits** [reply]
Yes, I understand that but I am talking about WebSubs (previously known as PubSubHubBub). 

Github docs say that `GitHub can also serve as a PubSubHubbub hub for all repositories.`
https://docs.github.com/en/rest/webhooks?apiVersion=2022-11-28#pubsubhubbub

I will be trying to make a POC, unless someone has already done this

**Reply by joecotton-wk** [reply]
@BoxedFruits how did your POC come out? were you able to get it to work?

**Reply by smcgivern** [reply]
@joecotton-wk I tried this out and it seems subscribing just creates a webhook on the repo, so you still need permissions to do that.

For one of my own repos, I did this and then saw the hook under Settings -> Webhooks. For a public repo that I don't own, like kubernetes/kubernetes, I get:

```json
{
  "message": "No repository found for hub.topic: \"https://github.com/kubernetes/kubernetes/events/push\"",
  "documentation_url": "https://docs.github.com/rest/reference/repos#pubsubhubbub"
}
```

The documentation URL also doesn't work 😬

**Reply by qaylinmoore** [reply]
Hi ! Talking in from the distant future. Did anybody get this working? I can't find any references to either WebSubs or pubsubhubbub on their docs?

**Comment by nimmiwe** [comment]
Correct — GitHub only lets repo owners or collaborators configure webhooks. If the repo isn’t yours, you can’t directly receive webhook events like releases.
Instead, ask the repo owner to add your webhook, or poll their releases API periodically:
https://api.github.com/repos/OWNER/REPO/releases/latest.

**Comment by puteulanus** [comment]
Some unverified ideas.

It seems that subscribing to custom events via the watch button can get email notifications when a repo release.

Using services like Cloudflare's Email Workers can convert these emails into API calls.

**Reply by GottZ** [reply]
ye that idea came to my mind too..
after all, the internet runs on email and excel spreadsheets.. and that is not even far fetched.

**Comment by alienryes** [comment]
Old topic, I know, but as a solution, I wrote a Komodo action that checks the 'latest' release version number from the Github releases API against a Komodo variable. If the release number is higher than the variable, a build is triggered and the variable is then updated to the new version number. I'm  running this on a schedule to check once every couple of days. Still testing at the moment with the various configurations of build numbers but so far, so good.
